service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/ManageOrdersEventBuilder.java

```diff
@@ -16,17 +22,30 @@
 @RequiredArgsConstructor(onConstructor_ = {@Autowired})
 public class ManageOrdersEventBuilder {
     private final SealedOrderHistoryService historyService;
+    private final List<AmendedOrderFinder<? extends AmendableOrder>> finders;
 
-    public Optional<GeneratedOrderEvent> build(CaseData caseData, CaseData caseDataBefore) {
+    public Optional<ManageOrdersEvent> build(CaseData caseData, CaseData caseDataBefore) {
         List<Element<GeneratedOrder>> currentOrders = caseData.getOrderCollection();
         List<Element<GeneratedOrder>> oldOrders = caseDataBefore.getOrderCollection();
+        GeneratedOrder lastGeneratedOrder = historyService.lastGeneratedOrder(caseData);
 
         if (!isAmendedOrder(currentOrders, oldOrders)) {
-            GeneratedOrder lastGeneratedOrder = historyService.lastGeneratedOrder(caseData);
             return Optional.of(new GeneratedOrderEvent(caseData, lastGeneratedOrder.getDocument()));
         }
 
-        return Optional.empty();
+        Optional<? extends AmendableOrder> order = finders.stream()
+            .map(finder -> finder.findOrderIfPresent(caseData, caseDataBefore))
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .findFirst();
```

Comment:
You are calling .get() in the next line without checking that an optional is present which could result in an NPE. One optional to solve this is you could add an extra map to the order chain to map it to the event

Reply:
somthing like
```java
        return finders.stream()
            .map(finder -> finder.findOrderIfPresent(caseData, caseDataBefore))
            .filter(Optional::isPresent)
            .map(Optional::get)
            .findFirst()
            .map(order -> new AmendedOrderEvent(
                    caseData, order.getDocument(), order.getAmendedOrderType(), order.getSelectedOthers()
                )
            );
```